orchestra/google_apps/service.py

```diff
@@ -129,3 +129,47 @@ def delete_folder(self, folder_id):
         except errors.HttpError as error:
             logger.exception('An error occured: %s', error)
             return None
+
+    def list_folder(self, folder_id):
+        """Retrieve a list of File resources.
+
+        Args:
+            folder_id: ID of parent folder to list.
+        Returns:
+            List of File resources.
+        """
+        service = self._service
+        result = []
+        page_token = None
+        while True:
```

Comment:
1. this is a direct copy-paste from https://developers.google.com/drive/v2/reference/files/list

2. The `while True:` doesn't implement retries, it implements paging if there are more than 100 files in the directory.

3. This code actually bails and returns `None` on the first failure, see the `except` clause below.